packages/server/src/fhir/search.ts

```diff
@@ -218,14 +228,25 @@ interface GetSelectQueryForSearchOptions extends GetBaseSelectQueryOptions {
 }
 function getSelectQueryForSearch<T extends Resource>(
   repo: Repository,
-  searchRequest: SearchRequest<T>,
+  searchRequest: SearchRequestWithCountAndOffset<T>,
   options?: GetSelectQueryForSearchOptions
 ): SelectQuery {
   const builder = getBaseSelectQuery(repo, searchRequest, options);
   addSortRules(builder, searchRequest);
-  const count = searchRequest.count as number;
+  const count = searchRequest.count;
   builder.limit(count + (options?.limitModifier ?? 1)); // Request one extra to test if there are more results
-  builder.offset(searchRequest.offset ?? 0);
+  if (searchRequest.offset > 0) {
+    builder.offset(searchRequest.offset);
+  }
+  if (searchRequest.cursor) {
```

Comment:
Should this be an `else if`?  Would we ever want someone combining these?

Reply:
Yeah... I couldn't decide which one should have priority.

I suppose if both are set, that should be an error?

Reply:
Made it an error to use offset>0 and cursor.

✅